src/squlearn/qnn/base_qnn.py

```diff
@@ -108,11 +116,132 @@ def fit(self, X: np.ndarray, y: np.ndarray, weights: np.ndarray = None) -> None:
             y: Labels
             weights: Weights for each datapoint
         """
-        self.param = self.param_ini.copy()
-        self.param_op = self.param_op_ini.copy()
+        self._param = self._param_ini.copy()
+        self._param_op = self._param_op_ini.copy()
         self._is_fitted = False
         self._fit(X, y, weights)
 
+    def get_params(self, deep: bool = True) -> dict:
+        """
+        Returns a dictionary of parameters for the current object.
+
+        Parameters:
+            deep: If True, includes the parameters from the base class.
+
+        Returns:
+            dict: A dictionary of parameters for the current object.
+        """
+        # Create a dictionary of all public parameters
+        params = {key: value for key, value in self.__dict__.items() if not key.startswith("_")}
+
+        if deep:
+            params.update(self._feature_map.get_params())
+            if isinstance(self._operator, list):
+                for i, oper in enumerate(self._operator):
+                    oper_dict = oper.get_params()
+                    for key, value in oper_dict.items():
+                        if key != "num_qubits":
+                            params[f"op{i}__{key}"] = value
+            else:
+                for key, value in self._operator.get_params().items():
+                    if key != "num_qubits":
+                        params[f"operator__{key}"] = value
+        return params
+
+    def set_params(self: BaseQNN, **params) -> BaseQNN:
```

Comment:
Short question: Why are the feature map and the operator are stored and treated separately from the QNN? Aren’t this parameter showing up twice? get_params and set_params of the QNN automatically update the featuremap and the operator stored within the QNN and reinitialize the QNN if necessary. Not sure, why it is necessary to change (or even store) the operator and the feature map in the baseclass.

Reply:
You're right. This way it's way easier. I changed it accordingly.